it/src/main/java/com/google/cloud/teleport/it/datastream/DatastreamResourceManager.java

```diff
@@ -0,0 +1,392 @@
+        ConnectionProfile.Builder connectionProfileBuilder = ConnectionProfile.newBuilder();
+        JDBCSource.SourceType type = source.type();
+
+        try {
+            switch (type) {
+                case MYSQL:
+                    MysqlProfile.Builder mysqlProfileBuilder = MysqlProfile.newBuilder();
+                    mysqlProfileBuilder.setHostname(source.hostname()).setUsername(source.username()).setPassword(source.password()).setPort(source.port());
+                    connectionProfileBuilder.setMysqlProfile(mysqlProfileBuilder);
+                    break;
+                case ORACLE:
+                    OracleProfile.Builder oracleProfileBuilder = OracleProfile.newBuilder();
+                    oracleProfileBuilder.setHostname(source.hostname()).setUsername(source.username()).setPassword(source.password()).setPort(source.port());
+                    connectionProfileBuilder.setOracleProfile(oracleProfileBuilder);
+                    break;
+                case POSTGRESQL:
+                    PostgresqlProfile.Builder postgresqlProfileBuilder = PostgresqlProfile.newBuilder();
+                    postgresqlProfileBuilder.setHostname(source.hostname()).setUsername(source.username()).setPassword(source.password())
+                            .setPort(source.port()).setDatabase(((PostgresSource) source).getDatabase());
+                    connectionProfileBuilder.setPostgresqlProfile(postgresqlProfileBuilder);
+                    break;
+            }
+
+            ConnectionProfile connectionProfile = connectionProfileBuilder
+                    .setDisplayName(connectionProfileId)
+                    .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance())
+                    .build();
+            CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                    .setParent(LocationName.of(projectId, location).toString())
+                    .setConnectionProfile(connectionProfile)
+                    .setConnectionProfileId(connectionProfileId)
+                    .build();
+            ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+            createdConnectionProfileIds.add(connectionProfileId);
+
+            LOG.info("Successfully created JDBC Source Connection Profile {} in project {}.", connectionProfileId, projectId);
+            return reference;
+        } catch (ExecutionException | InterruptedException e) {
+            throw new DatastreamResourceManagerException("Failed to create JDBC source connection profile. ", e);
+        }
+    }
+
+    public synchronized SourceConfig buildSourceConfig(String sourceConnectionProfileId, JDBCSource source) {
+        checkArgument(source.getAllowedTables() == null && source.getAllowedTables().isEmpty(),
+                "JDBC source allowed tables can not be null or empty");
```

Comment:
just saw the below comments on `config()`. This might not be relevant after the recommended changes in below comment

Reply:
As you said, this may not be relevant, but I was going off the fact that it is an optional setter method in the `JDBCSource.Builder` which would imply the `allowedTables` is optional. I would make that variable required in the builder's constructor should it stick around after your next refactor. If it gets factored out, then disregard.

Reply:
I think it could remain optional, and if it is not specified, it could default to all tables. Lmk what you think, or if that is even possible.